### PR TITLE
WX-1461 Remove `womtool upgrade` command

### PR DESCRIPTION
**Pull request by aednichols**
Also removes large chunks of the WDL writer that were only used for very specific error messages.

### CODE REVIEW

File: wdl/transforms/new-base/src/main/scala/wdl/transforms/base/wdlom2wom/CommandPartElementToWomCommandPart.scala [hunk 1]
```diff
@@ -57,8 +57,9 @@ case class WdlomWomStringCommandPart(stringCommandPartElement: StringCommandPart
 
 case class WdlomWomPlaceholderCommandPart(attributes: PlaceholderAttributeSet, expression: WdlomWomExpression)
     extends CommandPart {
-  def attributesToString: String = attributes.toWdlV1
+  private def attributesToString: String = attributes.toWdlV1
   // Yes, it's sad that we need to put ${} here, but otherwise we won't cache hit from draft-2 command sections
+  // Re-writes e.g. `echo $((5 + ~{x.a}))` commands to `echo $((5 + ${x.a}))` so they evaluate to equal
   override def toString: String = "${" + s"$attributesToString${expression.expressionElement.toWdlV1}" + "}"
```
Comment (aednichols) [inline wdl/transforms/new-base/src/main/scala/wdl/transforms/base/wdlom2wom/CommandPartElementToWomCommandPart.scala:63]: I'm not sure this is worth it or if I even agree with the original decision, but adjudicating that seemed out of scope for this PR. It does mean we have to retain the ability to write any expression element.

File: wdl/transforms/new-base/src/main/scala/wdl/transforms/base/wdlom2wom/CommandPartElementToWomCommandPart.scala [hunk 2]
```diff
@@ -18,7 +18,7 @@ import wdl.model.draft3.graph.ExpressionValueConsumer.ops._
 import wdl.model.draft3.graph.expression.WomExpressionMaker.ops._
 import wdl.transforms.base.linking.expression._
 import wdl.transforms.base.wdlom2wdl.WdlWriter.ops._
-import wdl.transforms.base.wdlom2wdl.WdlWriterImpl._
+import wdl.transforms.base.wdlom2wdl.WdlWriterImpl.{expressionElementWriter, placeholderAttributeSetWriter}
```
Comment (aednichols) [inline wdl/transforms/new-base/src/main/scala/wdl/transforms/base/wdlom2wom/CommandPartElementToWomCommandPart.scala:21]: This is a workflow thing that made it easier for me to see what is really being used or not.

File: wdl/transforms/draft3/src/test/scala/wdl/draft3/transforms/wdlom2wdl/WdlomToWdlFileSpec.scala [hunk 3]
```diff
@@ -1,63 +0,0 @@
...
-          val newModel = (stringToAst andThen wrapAst andThen astToFileElement).run(
-            FileStringParserInput(wdlModel.toWdlV1, file.name)
-          )
-
-          // Scala case class deep equality is so nice here
-          newModel.map(stripSourceLocations) shouldEqual model.map(stripSourceLocations)
-        case Left(_) => fail("Could not load original")
-      }
-    }
-  }
```
Comment (aednichols) [inline wdl/transforms/draft3/src/test/scala/wdl/draft3/transforms/wdlom2wdl/WdlomToWdlFileSpec.scala:62]: I remain proud of this code, which...
1. parses a WDL on disk to classes
2. writes it out with the WDL writer
3. re-parses the written-out version
4. compares the new classes with those in step (1)

**Review by jgainerdewar — APPROVED** [review]
So much red, you love to see it.

**Review by THWiseman — APPROVED** [review]

**Pull request merged**